src/Campaigns/Migrations/Donations/AddCampaignId.php

```diff
@@ -86,7 +104,56 @@ public function run()
                     ->insert($donationMeta, ['%d', '%s', '%d']);
             }
         } catch (DatabaseQueryException $exception) {
-            throw new DatabaseMigrationException("An error occurred while adding campaign ID to the donation meta table", 0, $exception);
+            throw new DatabaseMigrationException("An error occurred while adding campaign ID to the donation meta table",
+                0, $exception);
+        }
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getItemsCount(): int
+    {
+        return $this->query()->count();
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getBatchItemsAfter($lastId): ?array
+    {
+        $items = $this->query()
+            ->select('ID')
+            ->where('ID', $lastId, '>')
+            ->orderBy('ID')
+            ->limit($this->getBatchSize())
+            ->getAll();
+
+        if ( ! $items) {
+            return null;
         }
+
+        return [
+            min($items)->ID,
+            max($items)->ID,
+        ];
```

Comment:
It's faster to use a subquery and return only the first and last values:
```sql
SELECT MIN(id) AS first_id, MAX(id) AS last_id
FROM (
    SELECT id
    FROM posts
    WHERE id > 200
    ORDER BY id ASC
    LIMIT 100
) AS batch;
```

Reply:
Any thoughts on this, @alaca?

Reply:
Love it, I just have to implement it 😁

Reply:
Hmm, I don't see a reason to use the subquery here. We can simplify this by doing this:

```sql
SELECT MIN(id) AS first_id, MAX(id) AS last_id
    FROM posts
    WHERE id > 200
    ORDER BY id ASC
    LIMIT 100
```

I think it should return the same result... I'll check

Reply:
Yep, it does work the same

Reply:
Resolved https://github.com/impress-org/givewp/pull/7755/commits/62d5a167e6954e8aeb0465bbfad53b0c34438c5c

Reply:
Nice! I don't know why I thought it would need that. 😆